Functions/Solution.cs

```diff
@@ -15,75 +18,200 @@ internal class Solution
+                    appearsOnce = nums[i];
+                    appearseOnceCounter++;
+                }
+            }
+
+            if (appearseOnceCounter == 1)
+                return appearsOnce;
+
+            return -1;
         }
 
-        public static void Assignment8(bool[,] board, int rows, int cols)
+        public static void Assignment8(int[,] board, int rows, int cols)
         {
-            /*
-             * TODO: Write code here
-             */
+            int[,] newBoard = GameOfLife.UpdatedBoard(board, rows, cols);
+            
+            // Prints the output of game of life.
+            Console.WriteLine('{');
+            for (int row = 0; row < rows; row++)
+            {
+                Console.Write('{');
+                for (int col = 0; col < cols; col++)
+                {
+                    if (col == cols - 1)
+                        Console.Write(newBoard[row, col]);
+                    else
+                        Console.Write(newBoard[row, col] + ", ");
+                }
+                Console.WriteLine('}');
+            }
+            Console.WriteLine('}');
         }
 
         public static int Assignment9()
         {
-            /*
-             * TODO: Write code here
-             */
+            // 90% density of jumping numbers.
+            int min = 21780;
+            int max = 10000000;
+            int current = BouncingNumbers.Half(min, max);
+            double temp = BouncingNumbers.AmountOfBouncingNums(current) / 0.99;
+
+            while (temp != current)
+            {
+                if (temp < current)
+                    min = current;
+                else
+                    max = current;
+
+                current = BouncingNumbers.Half(min, max);
+                temp = BouncingNumbers.AmountOfBouncingNums(current) / 0.99;
+            }
+            return current;
         }
         #endregion
 
         #region Bonus solutions
         public static int Bonus1()
         {
-            /*
-             * TODO: Write code here
-             */
+            string[] vocab = {"One", "two", "three", "four", "five", "six", "seven", "eight", "nine",
+                "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen",
+                "eighteen", "nineteen", "twenty", "thirty", "forty", "fifty", "sixty", "seventy",
+                "eighty", "ninety"};
+
+            int letterCount = 0;
+
+            int oneToNine = Helper.CountLettersInRange(0, 9, vocab);
+            int tenToNineteen = Helper.CountLettersInRange(9, 19, vocab);
+            int twentyToNinety = Helper.CountLettersInRange(19, vocab.Length, vocab);
+
+            letterCount += oneToNine * 9;
+            letterCount += tenToNineteen;
+            letterCount += twentyToNinety * 10;
+            
+            return letterCount;
         }
 
-        public static int Bonus2(uint[] hightMap)
+        public static int Bonus2(int[] hightMap)
```

Comment:
This does not look like the logic we discussed a few weeks ago.
In addition, this is not trivial and has no comments.

Reply:
I changed the code a little. I think now it's more like the logic we discussed before.